### PR TITLE
93 multiple drops

### PR DESCRIPTION
**Pull request by BerinikeTech**

### CODE REVIEW

File: blockchain/contracts/NFTtheWorld.sol [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            _numberOfNFTs <= maxNumberOfNFTsToBuy[_dropHash],
+            "Sorry, you can't reserve more that 5% of the NFTs."
+        );
+        // We have to make sure that not more NFTs get reserved by users than we have NFTs available
+        require(
+            _numberOfNFTs <= availableNFTs[_dropHash].length - reservedNFTsCount[_dropHash],
+            "Sorry, not enough NFTs left for your request"
+        );
+        reservedNFTsCount[_dropHash] += _numberOfNFTs;
+        nftReservations[msg.sender] = _numberOfNFTs;
```
Comment (dominic-heil) [inline blockchain/contracts/NFTtheWorld.sol:94]: The number of reservations of a user is overwritten with each new join. I.e. if I create a drop, join it 5 times, create a new drop, without dropping the old one, join the new one once, then I have only 1 nftreservation. This likely is unexpected behavior.

Comment (BerinikeTech) [reply blockchain/contracts/NFTtheWorld.sol:94]: Good catch, will fix :)

**Review by steinkohl — APPROVED** [review]
I think i played everything through - LGTM

File: blockchain/contracts/NFTtheWorld.sol [hunk 2]
```diff
@@ -108,20 +117,29 @@ contract NFTtheWorld {
...
         }
     }
 
     // This function lets a user buy her reserved NFTs (one at a time)
     //TODO: Think about a time span during which the reserved NFTs have to be bought
-    function buyNFT(uint256 _price, uint256 _nftHash, uint256 _dropHash) public payable {
+    function buyNFT(
+        uint256 _price,
+        uint256 _nftHash,
+        uint256 _dropHash
```
Comment (dominic-heil) [inline blockchain/contracts/NFTtheWorld.sol:141]: Just curious, why do you use underscore variables?

**Review by dominic-heil — APPROVED** [review]
LGTM! Small side note: `generateRandomNumber` does not seem very random. I generated 4 hashes last night in one drop and got two times the same hash. But I think we do not need this method later anyways, so all good :D

**Comment by steinkohl** [comment]
> LGTM! Small side note: `generateRandomNumber` does not seem very random. I generated 4 hashes last night in one drop and got two times the same hash. But I think we do not need this method later anyways, so all good :D

I think we can use the timestamp as well, can't we?

File: blockchain/contracts/NFTtheWorld.sol (hunk 2 above)
Comment (BerinikeTech) [reply blockchain/contracts/NFTtheWorld.sol:141]: It's a common practice to use underscore for the function parameters in solidity :)

**Pull request merged**